richard-kendra/server.js

```diff
@@ -0,0 +1,195 @@
+  let SQL = `
+    INSERT INTO articles(title, author, author_url, category, published_on, body)
+    VALUES ($1, $2, $3, $4, $5, $6);
+  `;
+
+  let values = [
+    request.body.title,
+    request.body.author,
+    request.body.author_url,
+    request.body.category,
+    request.body.published_on,
+    request.body.body
+  ]
+
+  client.query( SQL, values )
+    .then(function() {
+      response.send('insert complete')
+    })
+    .catch(function(err) {
+      console.error(err);
+    });
+});
+
+app.put('/articles/:id', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This also matches #3 & #5 in the diagram, since it queries the DB and responds to the client. It's used by Article.updateRecord to update a record in the DB. This matches the 'U' in CRUD, since it's updating the record.
+
+  let SQL = `UPDATE articles
+  SET
+    title=$1,
+    author=$2,
+    author_url=$3,
+    category=$4,
+    published_on=$5,
+    body=$6
+  WHERE article_id=$7;`;
+
+  let values = [
+    request.body.title,
+    request.body.author,
+    request.body.author_url,
+    request.body.category,
+    request.body.published_on,
+    request.body.body,
+    request.params.id
+  ];
+
+  client.query( SQL, values )
+    .then(() => {
+      response.send('update complete')
+    })
+    .catch(err => {
+      console.error(err);
+    });
+});
+
+app.delete('/articles/:id', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This matches #3 and #5 in the diagram. It's making a request to the DB to delete the article with the given id and responds to the client. This is used by the method Article.deleteRecord() in article.js. This matches the 'D' in CRUD, (destroy a record).
+
+  let SQL = `DELETE FROM articles WHERE article_id=$1;`;
+  let values = [request.params.id];
+
+  client.query( SQL, values )
+    .then(() => {
+      response.send('Delete complete')
+    })
+    .catch(err => {
+      console.error(err);
+    });
+});
+
+app.delete('/articles', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This also matches #'s 3 and 5 in the diagram since it queries the DB and sends a response to the client. It's used by Article.truncateTable to delete all the articles. This corresponds to the 'D' in CRUD, destroy a table.
```

Comment:
This will be the as the one above. Response is being sent back to the server, not to the user.